find_max.rb

```diff
@@ -0,0 +1,19 @@
+
+# Find the maximum element between index 0 and max_index, exclusive
+
+def find_max(array, max_index)
```

Comment:
Do we want to change this function name to something more descriptive? Maybe 'find_max_value', or 'find_max_array_value'?